python/turbodbc/cursor.py

```diff
@@ -122,17 +123,34 @@ def fetchmany(self, size=None):
         return [row for row in islice(self, size)]
 
     def fetchallnumpy(self):
+        from numpy.ma import concatenate
+        batches = list(self._numpy_batch_generator())
+        column_names = [description[0] for description in self.description]
+        return OrderedDict(zip(column_names, [concatenate(column) for column in zip(*batches)]))
+
+    def fetchnumpybatches(self):
+        batchgen = self._numpy_batch_generator()
+        column_names = [description[0] for description in self.description]
+        for next_batch in batchgen:
+            yield OrderedDict(zip(column_names, next_batch))
+
+    def _numpy_batch_generator(self):
         self._assert_valid_result_set()
         if _has_numpy_support():
             from turbodbc_numpy_support import make_numpy_result_set
-            numpy_result_set = make_numpy_result_set(self.impl.get_result_set())
-            column_names = [description[0] for description in self.description]
-            columns = zip(column_names,
-                          [_make_masked_array(data, mask) for data, mask in numpy_result_set.fetch_all()])
-            return OrderedDict(columns)
         else:
             raise Error("turbodbc was compiled without numpy support. Please install "
                         "numpy and reinstall turbodbc")
+        numpy_result_set = make_numpy_result_set(self.impl.get_result_set())
+        firstrun = True
+        while True:
+            result_batch = numpy_result_set.fetch_next_batch()
+            batch_to_yield = _make_masked_arrays(result_batch)
```

Comment:
`batch_to_yield` is a bit of an unwieldy name, probably because the good name was already taken a line above. You could combine both lines into one:
```
batch = _make_masked_arrays(numpy_result_set.fetch_next_batch())
```